### PR TITLE
Availability: Fixes get account info retry logic to not go to secondary regions on 401(Forbidden)

### LINKED ISSUE #2496: wrong error when Aad access is unauthorized "The remote name could not be resolved myaccount-westus.documents.azure.com"
3.19-preview1
**Describe the bug**
when attempt to open a new connection using auth token credential provider i get
the remote name could not be resolved: 'myaccount-westus.documents.azure.com'

i have 2 regions only 1 is write, but both failed with same issue. when using preferredlocations.

**Expected behavior**
the cosmos client is returned

**Actual behavior**
an exception is thrown

**Environment summary**
SDK Version: 3.19-preview1
OS Version Windows latest

**Additional context**
 at Microsoft.Azure.Cosmos.Routing.GlobalEndpointManager.GetAccountPropertiesHelper.<GetAccountPropertiesAsync>d__8.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.GatewayAccountReader.<InitializeReaderAsync>d__6.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.CosmosAccountServiceConfiguration.<InitializeAsync>d__36.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.DocumentClient.<InitializeGatewayConfigurationReaderAsync>d__302.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.DocumentClient.<GetInitializationTaskAsync>d__96.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.DocumentClient.<EnsureValidClientAsync>d__156.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.DocumentClient.<GetPartitionKeyRangeCacheAsync>d__79.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.ContainerCore.<GetFeedRangesAsync>d__35.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.ClientContextCore.<RunWithDiagnosticsHelperAsync>d__38`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.ClientContextCore.<OperationHelperWithRootTraceAsync>d__29`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.CosmosClient.<InitializeContainerAsync>d__61.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Cosmos.CosmosClient.<CreateAndInitializeAsync>d__15.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Microsoft.CaseManagement.Common.Providers.DocumentDb.DbConnectionManager.<CreateAndInitializeAsync>d__10.MoveNext() in E:\repos\MSEG.CMC.CCM.Services.CaseManagement\Product\Source\Microsoft.CaseManagement.Provider.DocumentDb\DbConnectionManager.cs:line 117

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

This fixes a bug where if a user gets a 401(Forbidden) exception the SDK will try to go to the secondary regions. It will also now throw an aggregate exception if there are multiple failures instead of just throwing the last. This needed because it's possible the last failure maybe different then the cause of the other failures. 

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #2496

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 1]
```diff
@@ -190,12 +190,17 @@ public async Task<AccountProperties> GetAccountPropertiesAsync()
                     tasksToWaitOn.Remove(completedTask);
                 }
 
-                if (this.LastTransientException == null)
+                if (!this.TransientExceptions.Any())
                 {
                     throw new ArgumentException("Account properties and NonRetriableException are null and there is no LastTransientException.");
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:195]:
```suggestion
                    throw new ArgumentException("Account properties and NonRetriableException are null and there are no TransientExceptions.");
```

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 2]
```diff
@@ -217,12 +222,17 @@ private async Task<AccountProperties> GetOnlyGlobalEndpointAsync()
                     throw this.NonRetriableException;
                 }
 
-                if (this.LastTransientException != null)
+                if (!this.TransientExceptions.Any())
+                {
+                    throw new ArgumentException("Account properties and NonRetriableException are null and there is no LastTransientException.");
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:227]:
```suggestion
                    throw new ArgumentException("Account properties and NonRetriableException are null and there are no TransientExceptions.");
```

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 3]
```diff
@@ -217,12 +222,17 @@ private async Task<AccountProperties> GetOnlyGlobalEndpointAsync()
                     throw this.NonRetriableException;
                 }
 
-                if (this.LastTransientException != null)
+                if (!this.TransientExceptions.Any())
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:225]: This is a list, can we avoid the Linq call and just check the Count == 0?

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 4]
```diff
@@ -190,12 +190,17 @@ public async Task<AccountProperties> GetAccountPropertiesAsync()
                     tasksToWaitOn.Remove(completedTask);
                 }
 
-                if (this.LastTransientException == null)
+                if (!this.TransientExceptions.Any())
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:193]: This is a list, can we avoid the Linq call and just check the Count == 0?

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 5]
```diff
@@ -217,12 +222,17 @@ private async Task<AccountProperties> GetOnlyGlobalEndpointAsync()
...
-                if (this.LastTransientException != null)
+                if (!this.TransientExceptions.Any())
+                {
+                    throw new ArgumentException("Account properties and NonRetriableException are null and there is no LastTransientException.");
+                }
+
+                if (this.TransientExceptions.Count == 1)
                 {
-                    throw this.LastTransientException;
+                    throw this.TransientExceptions.First();
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:232]: Avoid Linq?

```suggestion
                    throw this.TransientExceptions[0];
```

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs [hunk 6]
```diff
@@ -190,12 +190,17 @@ public async Task<AccountProperties> GetAccountPropertiesAsync()
...
-                if (this.LastTransientException == null)
+                if (!this.TransientExceptions.Any())
                 {
                     throw new ArgumentException("Account properties and NonRetriableException are null and there is no LastTransientException.");
                 }
 
-                throw this.LastTransientException;
+                if (this.TransientExceptions.Count == 1)
+                {
+                    throw this.TransientExceptions.First();
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:200]: Avoid Linq?

```suggestion
                    throw this.TransientExceptions[0];
```

**Review by ealsur — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs (hunk 3 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:225]: What is the reason for avoiding the linq call?

File: Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs (hunk 4 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs:193]: This one got missed on the Linq removal?

**Review by ealsur — APPROVED** [review]

**Pull request merged**